### PR TITLE
Add PHP script engine

### PR DESCRIPTION
**Pull request by fviale**
- usual changes to include new script engine
 - in addition, modified XML pretty-print mechanism to avoid modifying CDATA section

### CODE REVIEW

File: app/scripts/proactive/config.js [hunk 1]
```diff
@@ -49,18 +49,19 @@ define(function () {
...
+            'Script/pre': [" ", "bash", "shell", "cmd", "kubernetes", "docker-compose", "dockerfile", "scalaw", "groovy", "javascript", "python", "cpython", "ruby", "perl", "php", "powershell", "R", "vbscript"],
+            'Script/task': [" ", "bash", "shell", "cmd", "kubernetes", "docker-compose", "dockerfile", "scalaw", "groovy", "javascript", "python", "cpython", "ruby", "perl", "php", "powershell", "R", "vbscript"],
+            'Script/post': [" ", "bash", "shell", "cmd", "kubernetes", "docker-compose", "dockerfile", "scalaw", "groovy", "javascript", "python", "cpython", "ruby", "perl", "php", "powershell", "R", "vbscript"],
             'Script/flow': [" ", "scalaw", "groovy", "javascript", "python", "cpython", "ruby", "powershell", "R"],
-            'Script/clean': [" ", "bash", "shell", "cmd", "kubernetes", "docker-compose", "scalaw", "groovy", "javascript", "python", "cpython", "ruby", "perl", "powershell", "R", "vbscript"]
+            'Script/clean': [" ", "bash", "shell", "cmd", "kubernetes", "docker-compose", "scalaw", "groovy", "javascript", "python", "cpython", "ruby", "perl", "powershell", "php", "R", "vbscript"]
```
Comment (mboussaa) [inline app/scripts/proactive/config.js]: "powershell", "php" => "php", "powershell"

File: app/scripts/proactive/view/xml/JobXmlView.js [hunk 2]
```diff
@@ -42,7 +42,43 @@ define(
...
+                cdataSections.push(middle)
+                var cdataStart = toParse.indexOf(cdataStartString)
+                var cdataEnd = toParse.indexOf(cdataEndString) + cdataEndString.length
+                cdataIndex++
+            }
+            accumulated += toParse
+            // pretty-print using vkbeautify, 'accumulated' variable contains xml with <![CDATA[index]]> patterns
+            var beautifiedXml = vkbeautify.xml(accumulated, beautifyDepth)
+
+            // after pretty-pring is done, replace back the original CDATA sections
```
Comment (mboussaa) [inline app/scripts/proactive/view/xml/JobXmlView.js]: pring => print

File: app/scripts/proactive/config.js (hunk 1 above)
Comment (fviale) [reply app/scripts/proactive/config.js]: actually, the list is not alphabetically sorted. Don't know how it should be

File: app/scripts/proactive/view/xml/JobXmlView.js [hunk 3]
```diff
@@ -42,7 +42,43 @@ define(
             // removing multiple \n before closing xml element tag
             jobRendering = jobRendering.replace(/\n+\s+>/g, '>\n');
             // indenting using vkbeautify
-            return vkbeautify.xml(jobRendering.trim(), 2);
+            return this.beautifyAndPreserveCDATA(jobRendering.trim(), 2);
+        },
+        // pretty-print xml and preserve CDATA sections from indentation changes
+        beautifyAndPreserveCDATA: function(text, beautifyDepth) {
+            var that = this;
```
Comment (mboussaa) [inline app/scripts/proactive/view/xml/JobXmlView.js]: where/how is this used `var that = this;` ?

Comment (fviale) [reply app/scripts/proactive/view/xml/JobXmlView.js]: this is typical javascript code, even if not used maybe now in this function, it is the general way to allow a subfunction call accessing the "this" object

**Pull request merged**